test/framework/src/test/java/org/elasticsearch/test/VersionUtilsTests.java

```diff
@@ -195,7 +199,12 @@ public void testGradleVersionsMatchVersionUtils() {
                 /* Gradle skips the current version because being backwards compatible
                  * with yourself is implied. Java lists the version because it is useful. */
                 .filter(v -> v != Version.CURRENT)
-                .map(v -> v.major + "." + v.minor + "." + v.revision)
+                /* Note that gradle does *not* skip alphas, betas, or rcs here even
+                 * though we don't have backwards compatibility for alphas, betas,
```

Comment:
I believe only the last beta/rc in a branch is unreleased so I think we're good there. I mean, it is technically possible for the release branch and the stable branch to have a beta but I don't think we'll do that so I'm not sure it is worth filtering it out.

I'm not sure why we don't have alphas in the Version.groovy. I mean, I know the regex doesn't match them, I just don't know why we treat them differently then betas. Either way, I can remove the word alpha here and make the filter just allow betas and rcs if you like.

I haven't changed the way gradle scrapes the versions. Did you want me to do that as part of this PR? We talked about wanting that "only the last beta/rc" behavior but I was under the impression we'd already implemented it. When I went to make this change it looks like we haven't and the reason our list lines up now is because we only pick up alphas.